### ISSUE TITLE
\PitonInputFile and path

### ISSUE THREAD

**Issue by danielezambelli** [fpantigny/piton#1]
With the new version the command:

`\PitonInputFile{\folder src/program.py}`

It no longer works.

Is it possible to allow this way to upload the file as well?

Thank you.

**Comment by fpantigny** [comment]
`\folder` is a LaTeX command that should be expanded to a string, isn't ?

**Comment by danielezambelli** [comment]
Yes, `\folder` contains the first part of the path.

**Comment by fpantigny** [comment]
You are right. The behavior of `\PitonInputFile` has changed in v. 2.2b of 2023-11-03. Now, the argument of `\PitonInputFile` is no longer expanded (in the TeX sens). I have changed because, previously, there were several characters (such as ~) that could not figure in the name of the file. My modification breaks the  backward compatibility. Sorry. However, for your situation, I have several suggestions.

1. Maybe you could try to use the key `path` added in version 2.2 of 2023-09-05.
2. You can force the expansion of the argument by writing `\expanded{\PitonInputFile{\folder /program.py}}`
3. You can write your own LaTeX macro for the previous construction. 
4. I can add in `piton` a new feature, for example a key to require an expansion of the argument of `\PitonInputFile` . Do you have a suggestion?

**Comment by danielezambelli** [comment]
Thank you for the rapid response.

`\expanded{\PitonInputFile{\folder /program.py}}`

It does not work, maybe because expanded inserts a cr after the 80th character (\folder contains a very long string).
The solution that works in my case is:

```
\expanded{\PitonOptions{path=\folder}}
\PitonInputFile{src/program.py}
```

but I am afraid that \expanded may not work with paths longer than 80 characters.

Thanks for the hints you sent me.

**Comment by danielezambelli** [comment]
> 4\. I can add in `piton` a new feature, for example a key to require an expansion of the argument of `\PitonInputFile` . Do you have a suggestion?

I have no suggestions because the previous behavior has always worked for me and I don't know what examples of failure this change has been tested on.

Thank you for your work.
